### PR TITLE
TASK: Documentation for node migrations

### PR DESCRIPTION
**Pull request by kdambekalns**
This adds documentation for node migrations (filters and transformations).

### CODE REVIEW

**Comment by dimaip** [comment]
:+1:

File: TYPO3.Neos/Documentation/References/NodeMigrations.rst [hunk 1]
```diff
@@ -0,0 +1,282 @@
+.. _`node-migrations`:
+
+Node Migrations
+===============
+
+Node migrations can be used to deal with renamed nodetypes and property names, set missing default values for
```
Comment (aertmann) [inline TYPO3.Neos/Documentation/References/NodeMigrations.rst]: node types

File: TYPO3.Neos/Documentation/References/NodeMigrations.rst [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+  down:
+    comments: 'No down migration available'
+
+Like all migrations the file should be placed in a package inside the ``Migrations/TYPO3CR`` folder where it will be picked
+up by the CLI tools provided with the content repository:
+
+- ``./flow node:migrationstatus``
+- ``./flow node:migrate``
+
+Use ``./flow help <command>`` to get detailed instructions. The migrationstatus command also prints a short description
```
Comment (aertmann) [inline TYPO3.Neos/Documentation/References/NodeMigrations.rst]: ``migrationstatus``

File: TYPO3.Neos/Documentation/References/NodeMigrations.rst [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+``dimensionValues`` (array)
+  An array of dimension names and values to set.
+``addDefaultDimensionValues`` (boolean)
+  Whether to add the default dimension values for all dimensions that were not given.
+
+AddNewProperty
+~~~~~~~~~~~~~~
+
+Add the new property and its value.
```
Comment (aertmann) [inline TYPO3.Neos/Documentation/References/NodeMigrations.rst]: s/the/a

File: TYPO3.Neos/Documentation/References/NodeMigrations.rst [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+  The value of the option "currentValuePlaceholder" (defaults to "{current}") will be used to include the current
+  property value into the new value.
+``search`` (string)
+  Search string to replace in current property value.
+``replace`` (string)
+  Replacement for the search string.
+``currentValuePlaceholder`` (string)
+  The value of this option (defaults to ``{current}``) will be used to include the current property value into the new
+  value.
```
Comment (aertmann) [inline TYPO3.Neos/Documentation/References/NodeMigrations.rst]: this information is not enough for me to understand how currentValuePlaceholder work, maybe an example would help?

File: TYPO3.Neos/Documentation/References/NodeMigrations.rst [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+RemoveNode
+~~~~~~~~~~
+
+Remove a given node (hard).
```
Comment (aertmann) [inline TYPO3.Neos/Documentation/References/NodeMigrations.rst]: how is this more hard than other transformations?

File: TYPO3.Neos/Documentation/References/NodeMigrations.rst [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+StripTagsOnProperty
+~~~~~~~~~~~~~~~~~~~
+
+Strip all tags on a given property.
+
+Options Reference:
+
+``property`` (string)
+  The name of the property to work on.
+
```
Comment (aertmann) [inline TYPO3.Neos/Documentation/References/NodeMigrations.rst]: some blank lines to clearly mark the section could be useful

File: TYPO3.Neos/Documentation/References/NodeMigrations.rst [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+- ``CreateNodes``
+- ``RemoveNode``
+- ``RemoveProperty``
+- ``RenameDimension``
+- ``RenameNode``
+- ``RenameProperty``
+- ``SetDimensions``
+- ``StripTagsOnProperty``
+
+They all implement the ``TYPO3\TYPO3CR\Migration\Transformations\TransformationInterface``, custom transformations
```
Comment (aertmann) [inline TYPO3.Neos/Documentation/References/NodeMigrations.rst]: reads a bit odd, maybe two separate sentences would be better?

File: TYPO3.Neos/Documentation/References/NodeMigrations.rst [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+comes with a number of filters:
+
+- ``DimensionValues``
+- ``IsRemoved``
+- ``NodeName``
+- ``NodeType``
+- ``PropertyNotEmpty``
+- ``Workspace``
+
+They all implement the ``TYPO3\TYPO3CR\Migration\Filters\FilterInterface``, custom filters can be developed against
```
Comment (aertmann) [inline TYPO3.Neos/Documentation/References/NodeMigrations.rst]: same here..

**Comment by aertmann** [comment]
a bunch of nit pick comments, feel free to adjust as you see fit

**Comment by kdambekalns** [comment]
Thanks. About the "blank lines", I doubt most people will read this in source form… or did I misunderstand something?

**Comment by kdambekalns** [comment]
Needs to go into 2.1 now

**Comment by aertmann** [comment]
@kdambekalns why only 2.1? isn't it relevant for 2.0 as well? technically even 1.2, but that's not worth the structure conflicts

**Comment by kdambekalns** [comment]
Because, lazy? Doesn't matter, I'll put it into 2.0. :)

File: TYPO3.Neos/Documentation/References/NodeMigrations.rst (hunk 5 above)
Comment (kdambekalns) [inline TYPO3.Neos/Documentation/References/NodeMigrations.rst]: Well, it doesn't mark it as removed. Took this from the transformation docblock… :)

**Comment by kdambekalns** [comment]
Fixed the comments, now closing and re-opening against 2.0.

**Pull request closed without merging**